### ISSUE TITLE
[Suggestion] Regular expressions tutorial

### ISSUE THREAD

**Issue by scpketer** [Monika-After-Story/MonikaModDev#5935]
I've been through Monika's Python tips (or guides? Not really sure), and an idea crossed my mind - what if Monika taught you how to use regexes?

I'm quite acquainted with those, so I think I could write an explaining topic about that (or multiple 'lessons' from basics to more advanced stuff, like with Python tips, but I don't really know how you guys made those) - but first I want to know if something like this will be okay to add, 'cause, well, most people think regexes are extremely complicated and so on...

If that's okay - great, I'll get to work then.

**Comment by Freshairkaboom** [comment]
I don't even know what regexes are yet, but I'd love anything Monika could teach me about using Python :D I'm probably going to learn it all before this gets added, but additions are just as much for newcomers as they are for those who are already playing the mod, so I'd welcome any help they'd get.

Also, when it comes to tips like that, then I think you don't need to pull out so much resources besides just making dialogue and labeling it, but I'm a newbie programmer and don't know what the code requires, so don't quote me on that :D

**Comment by ThePotatoGuy** [comment]
regex sounds good but we might need a prerequisite of more advanced string stuff like split/partition/splicing. ill double check this later.

the tips live in script-python so take a look at that to see how build the tips.

**Comment by scpketer** [comment]
> I don't even know what regexes are yet

"Basically, kind of a big deal" - an advanced string matching and substring extraction tool.

> we might need a prerequisite of more advanced string stuff like split/partition/splicing

Might as well bring these on.

**Comment by Freshairkaboom** [comment]
> 
> 
> > I don't even know what regexes are yet
> 
> An advanced string matching and substring extraction tool, but it might be really confusing to some.

Yeah...I'm a beginner, I don't understand what string matching is, I don't know what a substring is, extracting tool is kind of self explanatory though.

**Comment by scpketer** [comment]
Well, you can perform a simple string comparison with `"one" == "another"`, but this will only do for exact matches. 

With regular expressions you can perform something more complex like `"one" <something like> "another"` with `re.search(".*o.+e.*", "another")`

**Comment by Freshairkaboom** [comment]
You may as well have been speaking greek to me there. I only understood the first sentence.

**Comment by scpketer** [comment]
`re.search` returns search object, but if you don't need to extract matching occurences from strings you can just check if it's not None

**Comment by ThePotatoGuy** [comment]
Do you think you can write a prereq topic involving these string functions:
* `str.find` (`lfind`/`rfind`)
* `str.endswith` / `str.startswith`
* `str.isalumn` / `str.isalpha` / `str.isdigit` / `str.islower` / `str.isspace` / `str.istitle` / `str.isupper`
* `str.index` (`lindex`/`rindex`)
* `str.split` (`lsplit`/`rsplit`)
* `str.splitlines`
* `str.partition` (`lpartition`/ `rpartition`)
* string splicing using bracket notation `somestring[1:4]` (just getting a substring. setting the string should be saved for a more dedicated string replacement tip)

Basically all functions that can be used to find something in a string or to break a string up into pieces based on something. Understanding how to piece strings out using regular string functions would help in comparing what regexes can do.

**Comment by scpketer** [comment]
Sure, no problem, will do.

**Comment by scpketer** [comment]
Wait-wait, hold up - there is no `str.lfind` (as well as `str.lindex` and `str.lsplit`) neither in Python 3 nor Python 2.

**Comment by ThePotatoGuy** [comment]
ignore those, I assumed there were lversions of the rversions.

**Comment by scpketer** [comment]
Haha okay, there are just `r`-less versions of these methods
